packages/logger/src/sanitize.ts

```diff
@@ -1,31 +1,39 @@
-// Sanitize (but don't prettify) console output
+// Sanitize console output
+import stringify from 'fast-safe-stringify';
+export const SECRET = '****';
 
-import { LogOptions } from './options';
+type SanitizeOptions = {
+  stringify?: boolean;
 
-export const SECRET = '****';
+  sanitizePaths?: string[]; // unimplemented
+};
 
 // Node itself does a good job of circular references and functions
-const sanitize = (
-  item: any,
-  _options: Pick<LogOptions, 'sanitizePaths'> = {}
-) => {
-  // TODO what if the object contains functions?
-  if (typeof item !== 'string') {
+const sanitize = (item: any, options: SanitizeOptions = {}) => {
+  const maybeStringify = (o: any) =>
+    options.stringify === false ? o : stringify(o);
+  if (
+    Array.isArray(item) ||
+    (isNaN(item) &&
+      item &&
+      typeof item !== 'string' &&
+      // duck type an error object (this is not tight enough!)
+      !(item.stack && item.message))
```

Comment:
This is a problem: if an error object is passed to the log, it gets serialised to a string.

That's not actually the end of the world but it does break my tests, which expect to read an error object out of the log history, not a string.

The current code is not acceptable because the guard on the error object is too broad. For reasons I cannot grok right now `item instanceof Error` is false for the error object :thinking: 

Solutions:
1) Use json logging in the tests that want an error object. JSON logging won't stringify the object.
2) Make tests work on a string. It's fine!
3) Find a better way to write this guard case (or, work out why the error doesn't seem to actually be an Error)

I think I'm gonna do #1

Reply:
In fact I'm gonna do it now

Reply:
Ok 1 doesn't work because the json longer does actually serialize. Disappointing.

Reply:
Alright - what's going on here is that when the job, inside a SourceTextModule throws an error, we seem to get an error-like object back, not an actual Error. I think that's a detail of `node:vm` and I can't see any documentation about it - but the facts in the data are plain.

I've updated #143 with some extra notes. Basically we need to think a little bit more about how these errors get handled, propagated and reported, and this isn't the place to do it.

The runtime will now stringify this anomalous error, rather than logging it directly, which works fine if unit tests are updated a little.